src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1367,13 +1402,13 @@ protected void GenerateFindFirstChar()
                 Label returnFalseLabel = DefineLabel();
                 Label updatePosAndReturnFalse = DefineLabel();
 
-                Mvfldloc(s_runtextposField, _runtextposLocal);
-                Mvfldloc(s_runtextendField, _runtextendLocal);
-
-                // if (runtextend > runtextpos)
-                Ldloc(_runtextendLocal);
-                Ldloc(_runtextposLocal);
-                BleFar(returnFalseLabel);
+                if (minRequiredLength <= 0) // if minRequiredLength > 0, we already output a more stringent check
+                {
+                    // if (runtextend > runtextpos)
+                    Ldloc(_runtextendLocal);
+                    Ldloc(_runtextposLocal);
+                    BleFar(returnFalseLabel);
```

Comment:
Out of curiosity, how do you know to choose between far and not far jumps? You count the IL?

Reply:
Basically, yeah.  Or try a the short variety and change it when the runtime complains.